apps/src/studio/studio.js

```diff
@@ -3153,6 +3164,19 @@ Studio.setItemActivity = function (opts) {
     Studio.items.forEach(function (item) {
       if (item.className === opts.className) {
         item.setActivity(opts.type);
+
+        // For verifying success, record this combination of activity type and
+        // item type.
+
+        if (!Studio.trackedBehavior.setActivityRecord) {
+          Studio.trackedBehavior.setActivityRecord = [];
+        }
+
+        if (!Studio.trackedBehavior.setActivityRecord[opts.className]) {
+          Studio.trackedBehavior.setActivityRecord[opts.className] = [];
+        }
+
+        Studio.trackedBehavior.setActivityRecord[opts.className][opts.type] = true;
```

Comment:
I am being explicitly picky and checking that they have managed to actually set an activity character's behaviour.  I've opened another PR that is more forgiving: https://github.com/code-dot-org/code-dot-org/pull/4532